public/client.js

```diff
@@ -241,6 +241,19 @@ function cacheLoadImage (filename) {
   return imageCache[filename]
 }
 
+function lazyLoadImageFromCache (filename) {
+  return () => {
```

Comment:
Instead of returning an image we return a function that returns a promise that resolves to an image.